cmd/ingest/ingest.go

```diff
@@ -42,24 +57,162 @@ func main() {
 	var message Message
 
 	go func() {
-		for d := range broker.GetMessages(mq, config.Broker.Queue) {
-			log.Debugf("received a message: %s", d.Body)
-			err := json.Unmarshal(bytes.Replace(d.Body, []byte(`'`), []byte(`"`), -1), &message)
-			if err != nil {
-				log.Errorf("Not a json message: %s", err)
+		for delivered := range broker.GetMessages(mq, config.Broker.Queue) {
+			log.Debugf("Received a message: %s", delivered.Body)
+			if e := json.Unmarshal(delivered.Body, &message); e != nil {
+				log.Errorf("Not a json message: %s", e)
 			}
 
-			e := reflect.ValueOf(&message).Elem()
-			for i := 0; i < e.NumField(); i++ {
-				if e.Field(i).Interface() == "" || e.Field(i).Interface() == 0 {
-					log.Errorf("%s is missing", e.Type().Field(i).Name)
-					err = fmt.Errorf("%s is missing", e.Type().Field(i).Name)
+			fileID, ife := db.InsertFile(fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath), message.User)
+			if ife != nil {
+				log.Errorf("InsertFile failed, reason: %v", ife)
+				// This should really be hadled by the DB retry mechanism
+			}
+
+			file, rfe := storage.ReadFile(config.Inbox.Type, fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath))
+			if rfe != nil {
+				log.Errorf("Failed to open file: %s, reason: %v", fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath), rfe)
+				continue
+			}
+
+			fileSize, gfse := storage.GetFileSize(config.Inbox.Type, fmt.Sprintf("%s/%s", config.Inbox.Location, message.FilePath))
```

Comment:
Why are you creating new variables for `error` every time? You can just keep reusing `err` which you had in the beginning. No need in `e`, `ife`, `rfe`, `gfse`, etc.

Reply:
except that you get shadow declaration warnings for every time you repeat the declaration

Reply:
This is weird. I have just replaced all the errors with `err` in this method and IDE didn't highlight anything. Neither did linter complain. Can you give a particular example of the problematic usage of `err` in this method?

Reply:
If you have `err` on L66 you will get a shado declaration error on L111

`
go vet -vettool=$(which shadow) cmd/ingest/*
cmd/ingest/ingest.go:111:14: declaration of "err" shadows declaration at line 66
`

Reply:
First of all, in all the places where you have `_` before the `err`  on the left side you should use `=` instead of `:=` - this will reuse existing `err` and will not produce the shadowing warning. Secondly, this shadowing feature is experimental and disabled by default, reportedly having many false-positives. I don't know why would you enable it. Check this out, for example: https://github.com/golang/go/issues/19490  In third, shadowing is not necessarily bad even if it really happens. In the example above, having outer `err` from line 66 shadowed by the `inner` error at line 111 doesn't break anything. So, again, I'd rather disable this linter feature (since it's disabled by default).

Reply:
P.S. Also I don't know which IDE do you use, but in case if shadowing can really break something or have negative consequences, GoLand highlights it in the code. But if it's harmless shadowing, it doesn't.